vllm/model_executor/layers/quantization/kernels/MarlinLinearKernel.py

```diff
@@ -0,0 +1,128 @@
+from vllm import _custom_ops as ops
+from vllm.model_executor.layers.quantization.utils.marlin_utils import (
+    MARLIN_SUPPORTED_GROUP_SIZES, apply_gptq_marlin_linear,
+    check_marlin_supports_shape, marlin_is_k_full, marlin_make_empty_g_idx,
+    marlin_make_workspace, marlin_permute_scales, marlin_sort_g_idx,
+    query_marlin_supported_quant_types)
+from vllm.model_executor.parameter import (BasevLLMParameter,
+                                           permute_param_layout_)
+
+from .MPLinearKernel import *
+
+
+class MarlinLinearKernel(MPLinearKernel):
+
+    @classmethod
+    def get_min_capability(cls) -> int:
+        return 80
+
+    @classmethod
+    def can_implement(cls,
+                      c: MPLinearLayerConfig) -> Tuple[bool, Optional[str]]:
+        if c.zero_points:
+            return False, "Zero points currently not supported by "\
+                          " MarlinLinearKernel. Will be added when AWQMarlin "\
+                          "is migrated over to using MPLinearKernel backend"
```

Comment:
This won't affect awq_marlin for now since that backend is being left as-is right?

Reply:
Ya we haven't moved AWQ to use `choose_mp_linear_kernel` since we currently only have a repack kernel from AWQ to Marlin, really we should have:
```
standard_layout = {packed_dim: 0, input_dim: 0, output_dim: 1, packed_order: "contiguous}
AWQ -> standard_layout
GPTQ -> standard_layout
standard_layout -> Machete
standard_layout -> Marlin
```
right now we only have:
```
AWQ -> Marlin
GPTQ -> standard_layout
standard_layout -> Machete
standard_layout -> Marlin
```
so AWQ will have to be a separate PR

Reply:
Makes sense just wanted to confirm, thanks!